File: AdminArea/AdminArea.Web/Modules/Productcom/Productcom/ProductcomForm.cs
using Serenity.ComponentModel;
using System;

namespace AdminArea.Productcom.Forms;

[FormScript("Productcom.Productcom")]
[BasedOnRow(typeof(ProductcomRow), CheckNames = true)]
public class ProductcomForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string Image { get; set; }
    public int CategoryId { get; set; }
    public DateTime Date { get; set; }
}

File: AdminArea/AdminArea.Web/Modules/ServerTypes/Productcom/ProductcomForm.ts
import { StringEditor, DecimalEditor, ServiceLookupEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ProductcomForm {
    Name: StringEditor;
    Description: StringEditor;
    Price: DecimalEditor;
    Image: StringEditor;
    CategoryId: ServiceLookupEditor;
    Date: DateEditor;
}

export class ProductcomForm extends PrefixedContext {
    static readonly formKey = 'Productcom.Productcom';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductcomForm.init)  {
            ProductcomForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = ServiceLookupEditor;
            var w3 = DateEditor;

            initFormType(ProductcomForm, [
                'Name', w0,
                'Description', w0,
                'Price', w1,
                'Image', w0,
                'CategoryId', w2,
                'Date', w3
            ]);
        }
    }
}